website/source/api/search.html.md

```diff
@@ -0,0 +1,109 @@
+---
+layout: api
+page_title: Search - HTTP API
+sidebar_current: api-search
+description: |-
+  The /search endpoint returns matches for a given prefix and context
+---
+
+# Search HTTP API
+
+The `/search` endpoint returns matches for a given prefix and context, where a
+context can be jobs, allocations, evaluations, nodes, or deployments.
+Additionally, a prefix can be searched for within every context.
+
+| Method  | Path                         | Produces                   |
+| ------- | ---------------------------- | -------------------------- |
+| `POST`  | `/v1/search                 | `application/json`         |
+
+The table below shows this endpoint's support for
+[blocking queries](/api/index.html#blocking-queries) and
+[required ACLs](/api/index.html#acls).
+
+| Blocking Queries | ACL Required |
+| ---------------- | ------------ |
+| `NO`             | `none`       |
+
+### Parameters
+
+- `Prefix` `(string: <required>)` - Specifies the identifer against which
+  matches will be found. For example, if the given prefix were "a", potential
+  matches might be "abcd", or "aabb".
+- `Context` `(string: <required>)` - Defines the scope in which a search for a
+  prefix operates. Contexts can be: "jobs", "evals", "allocs", "nodes",
```

Comment:
"all" is actually just the variable name. It goes to empty string. We should probably just make the all context "all"

Reply:
Agreed. Maybe we can do that refactor in another PR? I'll keep this as an empty string for now.